### PR TITLE
Nightmare spawn event

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: Nightmares now have a chance to spawn via event.
/:cl:

Just a generic ghost spawn event for the pool since the antagonist is more or less functional now

### CODE REVIEW

File: code/modules/events/nightmare.dm [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/datum/round_event/ghost_role/nightmare/spawn_role()
+	var/list/candidates = get_candidates("alien", null, ROLE_ALIEN)
+	if(!candidates.len)
+		return NOT_ENOUGH_PLAYERS
+
+	var/mob/dead/selected = pick_n_take(candidates)
+
+	var/datum/mind/player_mind = new /datum/mind(selected.key)
+	player_mind.active = 1
```
Comment (MoreRobustThanYou) [inline code/modules/events/nightmare.dm]: TRUE

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by kevinz000** [comment]
**unzips floodlight**

File: code/modules/events/nightmare.dm (hunk 1 above)
Comment (kevinz000) [reply code/modules/events/nightmare.dm]: mrty catching headcoder in their mistakes ayyy

File: code/modules/events/nightmare.dm [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+/datum/round_event/ghost_role/nightmare
+	minimum_required = 1
+	role_name = "nightmare"
+
+/datum/round_event/ghost_role/nightmare/spawn_role()
+	var/list/candidates = get_candidates("alien", null, ROLE_ALIEN)
+	if(!candidates.len)
+		return NOT_ENOUGH_PLAYERS
+
+	var/mob/dead/selected = pick_n_take(candidates)
```
Comment (Cyberboss) [inline code/modules/events/nightmare.dm]: pick() is faster since you never use candidates again

File: code/modules/events/nightmare.dm [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+	var/list/spawn_locs = list()
+	for(var/obj/effect/landmark/xeno_spawn/L in GLOB.landmarks_list)
+		if(isturf(L.loc))
+			var/turf/T = L.loc
+			var/light_amount = T.get_lumcount()
+			if(light_amount < SHADOW_SPECIES_LIGHT_THRESHOLD)
+				spawn_locs += L.loc
+
+	if(!spawn_locs)
```
Comment (Cyberboss) [inline code/modules/events/nightmare.dm]: .len

File: code/modules/events/nightmare.dm [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+		message_admins("No valid spawn locations found, aborting...")
+		return MAP_ERROR
+
+	var/mob/living/carbon/human/S = new ((pick(spawn_locs)))
+	player_mind.transfer_to(S)
+	player_mind.assigned_role = "Nightmare"
+	player_mind.special_role = "Nightmare"
+	SSticker.mode.traitors |= player_mind
```
Comment (Cyberboss) [inline code/modules/events/nightmare.dm]: +=

File: code/modules/events/nightmare.dm [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	var/datum/mind/player_mind = new /datum/mind(selected.key)
+	player_mind.active = TRUE
+
+	var/list/spawn_locs = list()
+	for(var/obj/effect/landmark/xeno_spawn/L in GLOB.landmarks_list)
+		if(isturf(L.loc))
+			var/turf/T = L.loc
+			var/light_amount = T.get_lumcount()
+			if(light_amount < SHADOW_SPECIES_LIGHT_THRESHOLD)
+				spawn_locs += L.loc
```
Comment (Cyberboss) [inline code/modules/events/nightmare.dm]: += T

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Cyberboss** [comment]
Changelog

**Review by Cyberboss — APPROVED** [review]

**Comment by Supermichael777** [comment]
have they been made less lame?

**Comment by Armhulen** [comment]
no

**Comment by TechnoAlchemisto** [comment]
Headmin, Headcoder, and Design Lead; Watch out MSO, this guy is coming for you!

KorPhaeron - Yet again, you've dazzled me with your knack for programming, and your tenacity. The average person. Would have thought "I love these nightmare things, it's like horror movies (I like those) but I wish they were in the game more often! I wish I was smart, like KorPhaeron, and HotelBravoLima, my favorite administrator". Your problem solving skills are supernatural, and your judgement is an anomaly in this world that makes less and less sense by the day. I hope you're doing well, can't wait to see you in July at the reunion!

Sincerely,
TechnoAlchemist

PS - Did you change your name on my discord server? You don't seem to be there anymore and I can't seem to DM you. I hope your computer still works! xoxoxoxo

**Pull request merged**